src/MallardMessageHandlers/AuthenticationToken/AuthenticationTokenHandlerContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace MallardMessageHandlers
+{
+	/// <summary>
+	/// This represents the context for a <see cref="AuthenticationTokenHandler{TAuthenticationToken}"/>.
+	/// It's use mostly to help <see cref="AuthenticationTokenHandler{TAuthenticationToken}"/> handles thoses actions:
+	/// - Refreshing the token when expired.
+	/// - Detecting an expiration session.
+	/// By default, each <see cref="AuthenticationTokenHandler{TAuthenticationToken}"/> has its own context.
+	/// However, if there's multiple <see cref="AuthenticationTokenHandler{TAuthenticationToken}"/> (e.g. multiple endpoints),
+	/// it's strongly recommended that they're all shared the same context,
+	/// otherwise there's going to be synchronization issues when one of the endpoint either refresh the TAuthenticationToken or signal an expired session.
+	/// </summary>
+	public class AuthenticationTokenHandlerContext
```

Comment:
The name Context suggests that there are multiple contexts, yet the recommendation is to have 1 shared instance. I'm wondering if the name/pattern is correct.

Reply:
The handler definitely needs a reference to a shared semaphore. However, it seems like that the thing responsible for the lock should be the same thing that is responsible for the ownership of the token. That means that the IAuthenticationTokenProvider is probably the one responsible for the locking of the refresh process.

Reply:
Knowing that, I think `AuthenticationTokenHandlerContext` should be removed.

Reply:
they should only share the same context if they share the same authentication. That's the exact detail why we asked for this refactor. IF you need to authenticate against 2 different services then you should have 2 context, otherwise you'll only refresh one...

Reply:
but that would mean that you have 2 instances of `IAuthenticationTokenProvider` ?